File: merxly_backend/merxly.Domain/Entities/ProductAttribute.cs
namespace merxly.Domain.Entities
{
    public class ProductAttribute
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public int DisplayOrder { get; set; }
        public Guid ProductId { get; set; }

        // Navigation properties
        public Product Product { get; set; }
        public ICollection<ProductAttributeValue> ProductAttributeValues { get; set; } = new List<ProductAttributeValue>();
    }
}

File: merxly_frontend/src/types/models/productAttribute.ts
import type {
  CreateProductAttributeValueDto,
  ProductAttributeValueDto,
} from './productAttributeValue';

export interface CreateProductAttributeDto {
  name: string;
  displayOrder: number;
  productAttributeValues: CreateProductAttributeValueDto[];
}

export interface ProductAttributeDto {
  id: string;
  name: string;
  displayOrder: number;
  productAttributeValues: ProductAttributeValueDto[];
}

export interface ResponseUpdateAttributeItemDto {
  id: string;
  name: string;
  displayOrder: number;
}
